lib/events.js

```diff
@@ -182,55 +180,62 @@ function _addListener(target, type, listener, prepend) {
 
   events = target._events;
   if (events === undefined) {
-    events = target._events = Object.create(null);
-    target._eventsCount = 0;
+    events = target._events = new Map();
   } else {
     // To avoid recursion in the case that type === "newListener"! Before
     // adding it to the listeners, first emit "newListener".
-    if (events.newListener !== undefined) {
-      target.emit('newListener', type,
-                  listener.listener ? listener.listener : listener);
+    if (events.has('newListener')) {
+      target.emit('newListener', type, listener.listener || listener);
 
       // Re-assign `events` because a newListener handler could have caused the
       // this._events to be assigned to a new object
       events = target._events;
+      if (events === undefined)
+        events = target._events = new Map();
     }
-    existing = events[type];
+    existing = events.get(type);
+  }
+
+  if (type === 'error') {
+    events._hasErrorListener = true;
```

Comment:
I don't think we should be adding new, underscored properties to `EventEmitter` instances. Let's use symbols...

Reply:
I think the problem is that we need to get this from C++ so the alternative is a symbol that's declared in C++ and shared with JS? I'm not too familiar with those parts so open to feedback/corrections.

Reply:
@apapirovski Sorry, as you know I was on mobile, I probably should have given more context.

What I had in mind was having this tracking be specific to `Domain`s, not all `EventEmitter`s; you could add `.on('newListener');` and `.on('removeListener');` checks to keep track of whether an error listener is present, then update a property directly on the `Domain` object.

And, in case I wasn’t clear on it: That still means we’re doing invalid stuff in `ShouldAbortOnUncaughtException()`, getting properties isn’t allowed either way – it’s just going to work as long as we don’t invoke any runnable code, I guess. (But fixing that is definitely out of scope for this PR. :smile:)

Reply:
I agree.

Reply:
This is what happens when you almost lose your mind to JS not working in that chunk of C++ code and forget how to write code... 😆Thanks @addaleax, that's a much cleaner solution.